src/internals/strings/impl/length.ts

```diff
@@ -1,66 +1,42 @@
 // Original impl https://gist.github.com/sno2/7dac868ec6d11abb75250ce5e2b36041
 
 import { Add } from "../../numbers/impl/addition";
-import { Mul } from "../../numbers/impl/multiply";
+import { StringIterator as StrIter } from "./utils";
 
-type TCache = [string, 0[]];
+// Implementation of the following algorithm:
+//
+// function length(s, acc) {
+//   if (s === "") {
+//       return 0;
+//   } else if (s.indexOf(acc) === 0) {
+//       return acc.length + length(s.slice(acc.length), acc + acc);
+//   } else {
+//       return length(s, acc.slice(acc.length / 2));
+//   }
+// }
```

Comment:
I might be missing something, but the type-level algo below doesn't seem to match this JS implementation since parameters have different types (in this JS implementation it's 2 strings, whereas the tl implementation takes (string, number, iterator)).

Would you mind updating this comment?